lib/base/RestException.js

```diff
@@ -4,12 +4,12 @@ class RestException extends Error {
   constructor(response) {
     super('[HTTP ' + response.statusCode + '] Failed to execute request');
 
-    const body = response.body === 'string' ? JSON.parse(response.body) : response.body;
+    const body = typeof response.body === 'string' ? JSON.parse(response.body) : response.body;
     this.status = response.statusCode;
     this.message = body.message;
     this.code = body.code;
     this.moreInfo = body.more_info; /* jshint ignore:line */
-    this.detail = body.detail;
+    this.details = body.details === undefined ? null : body.details;
```

Comment:
Just `this.details = body.details;` is fine.

Reply:
@childish-sambino fixed.

Reply:
Tests are failing now.

Reply:
oops sorry, should be fixed now